src/mo_frontend/typing.ml

```diff
@@ -1440,15 +1442,20 @@ and infer_exp'' env exp : T.typ =
     T.Array (match mut.it with Const -> t1 | Var -> T.Mut t1)
   | IdxE (exp1, exp2) ->
     let t1 = infer_exp_promote env exp1 in
-    (try
-      let t = T.as_array_sub t1 in
+    begin match t1 with
+    | T.(Prim Blob) ->
```

Comment:
This only works when t1 is normalised, right? Better use `| t with sub t T.blob ->`

Reply:
@crusso is it guaranteed that `t1` is always normalised?

Reply:
Ok, I see, `T.promote` _does_ remove `Con`s. So this can stay.